src/main/java/org/openpnp/spi/PnpJobProcessor.java

```diff
@@ -1,4 +1,13 @@
 package org.openpnp.spi;
 
+import java.util.List;
+
+import org.openpnp.machine.reference.ReferencePnpJobProcessor.JobPlacement;
+
 public interface PnpJobProcessor extends JobProcessor {
+
+    public List<JobPlacement> getJobPlacementsById(String id);
+
+    public List<JobPlacement> getJobPlacementsById(String id, String status);
```

Comment:
This should take the enum value rather than a String for status. This enforces type safety.  So, just change it to `public List<JobPlacement> getJobPlacementsById(String id, JobPlacement.Status status);`